### ISSUE TITLE
SSH multiplexing doesn't work if multiple thread tries to connect to the same host

### ISSUE THREAD

**Issue by Hologos** [Hologos/tpcp#23]
Needs futher investigation. Here is what I think is happening:

If multiple threads tries to connect to the same host, the very first one detects that no master connection is created. So it creates one. But in meantime other threads detect that master connection is opened but it actually isn't fully initialized yet.

**Comment by Hologos** [comment]
Another type of error messages:

```
[28.08.2019 11:16:29] I: Program invoked as ./tpcp AAAK000001 BBB100 CCC200 CPY ADD
[28.08.2019 11:16:29] I: Loading ini file /nagpci/bsj-scripts/tpcp/system-definition.ini.
[28.08.2019 11:16:29] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:29] W: Control socket connect(/tmp/ssh-master-control-bbbadm@bbbci:22): No such file or directory
[28.08.2019 11:16:29] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:29] W: Control socket connect(/tmp/ssh-master-control-cccadm@ccc00:22): No such file or directory
[28.08.2019 11:16:30] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:30] W: Control socket connect(/tmp/ssh-master-control-aaaadm@aaa00:22): No such file or directory
[28.08.2019 11:16:30] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:30] W: Control socket connect(/tmp/ssh-master-control-aaaadm@aaa00:22): No such file or directory
[28.08.2019 11:16:30] W: ControlSocket /tmp/ssh-master-control-aaaadm@aaa00:22 already exists, disabling multiplexing
[28.08.2019 11:16:30] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:30] W: Master running (pid=941)
[28.08.2019 11:16:30] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:30] W: Master running (pid=941)
[28.08.2019 11:16:31] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:31] W: Master running (pid=872)
[28.08.2019 11:16:31] W: Pseudo-terminal will not be allocated because stdin is not a terminal.
[28.08.2019 11:16:31] W: Master running (pid=866)
```

**Comment by Hologos** [comment]
The other explanation is that both thread detect that no master connection is created to the same host. Both tries to create one. One succeeds, the other fails.

**Comment by Hologos** [comment]
Partially fixed with b10568f5ba3402560c3b154866674387cd3a8a7f and 77f0d0febe1d6ab8f04769f361152b6f1ddcd429 since most of the connections are created when discovering transport domains.

Still there will be some connections made on-demand (eg. copying from transport source).

**Comment by Hologos** [comment]
Commit a4bcc5f fixed last occurence of on-demand connection. Beware of #45 to not introduce it back.
